creusot/src/backend/clone_map/elaborator.rs

```diff
@@ -147,54 +154,40 @@ impl DepElab for LogicElab {
     ) -> Vec<why3::declaration::Decl> {
         assert!(matches!(
             dep,
-            Dependency::Item(_, _)
-                | Dependency::TyInvAxiom(_)
-                | Dependency::StructuralResolve(_)
-                | Dependency::ClosureSpec(_, _, _)
+            Dependency::Item(_, _) | Dependency::TyInvAxiom(_) | Dependency::ClosureSpec(_, _, _)
         ));
 
         // TODO: Fold into `term`, but requires first some sort of
         // handling for axioms
         if let Dependency::TyInvAxiom(ty) = dep {
-            return expand_ty_inv(elab, ctx, ty);
+            return expand_ty_inv_axiom(elab, ctx, ty);
```

Comment:
Any reason why you couldn't do that?

Reply:
I could not do what?

I guess you mean folding this into term, but I have to say I don't really see how this could work. `TyInvAxioms` means inserting a new *axiom*, while this piece of code inserts a new logical definition.